### DISCUSSION TITLE
A bit better password reset tool

### DISCUSSION THREAD

**Discussion by Kixunil** [btcpayserver/btcpayserver#2573] [Ideas / Feature Requests]
I've created a bit nicer [BTCPay password reset tool](https://github.com/Kixunil/btcpay-password-reset) which allows the user to set specific password for given account without having to create another account or do other funny things.

Feel free to use/copy/whatever. :)

**Comment by pavlenex** [comment]
@NicolasDorier @Kukks thoughts on adding this?

**Reply by Kukks** [reply]
Cool! Great tool. Maybe we could add a submodule in `contrib` folder for btcpayserver-docker + a compiled version?

**Comment by michnovka** [comment]
I would really appreciate a way to set passwords from an admin account on other accounts without the need of e-mail confirmation. Now I have to delete the old account and create it again.

**Reply by Kixunil** [reply]
My tool should work if you can use command line & ssh.

**Comment by Kukks** [comment]
I think we should offer:
* a direct password change from admin ui (maybe?)
* a way to unset password but force the end user to use a magic link to be
able to set a new password (can be sent via email)

**Reply by Kixunil** [reply]
My tool can reset admin password and doesn't require working SMTP. These alternatives are good, my tool is for other scenarios - mainly admin having SSH access but not web access.
